### DISCUSSION TITLE
Pushing internal service releases to repositories using the service / dependency

### DISCUSSION THREAD

**Discussion by jaredclarke** [mend/renovate-ce-ee#297] [Ideas]
Hello

I would like to use renovate-on-prem to push updates to downstream users of a dependency after a (gitlab) release event.

We have 3000 repositories that are being scanned so being able to jump the queue with those new internal releases makes a lot of sense for our workflow.

Has it previously been considered? or is it just a case of leave the updates till the scheduled job runs and that is good enough? 

thanks

**Comment by rarkins** [comment]
I understand your use case, but currently it's not supported. Renovate On-Prem does not keep the list of dependencies of each repo so would not have a way to know which ones to trigger, even if the API was added to receive the event.

**Reply by jaredclarke** [reply]
Is there a way to manually trigger a run against a single repository or add to the queue system?

**Reply by rarkins** [reply]
Not yet. A feature request for that exists: #49

**Reply by jaredclarke** [reply]
and https://github.com/mend/renovate-on-prem/issues/4 but both of these are >= 4 years old - what are the odds of implementation at this stage?

I was thinking about this last night and might implement a proxy service that intercepts all webhook events and transforms release events into multiple events for the downstream repositories - do you think that could work?

**Reply by rarkins** [reply]
It's a challenge for us to prioritize spending developer resources as it's a free product. The answer is that users wanting these features should become Mend customers and demand the feature that way.

I'm not sure that you'll be able to spoof webhooks because there is some validation which occurs.

**Reply by jaredclarke** [reply]
yes I understand the resourcing issue - given the move to a free-but-licensed product is there any appetite for a shift to OSS like renovatebot?

I'm happy to spend a bit of time and see what happens - it's something a little different from the day-to-day

**Reply by rarkins** [reply]
This project includes some proprietary code from Mend and is built from a monorepo which includes commercial products, so open sourcing would itself take a long effort

**Comment by jaredclarke** [comment]
thanks for the replies
